placeholders_test.go

```diff
@@ -0,0 +1,132 @@
+package txtdirect
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestParsePlaceholders(t *testing.T) {
+	tests := []struct {
+		url       string
+		requested string
+		expected  string
+	}{
+		{
+			"example.com{uri}",
+			"https://example.com/?test=test",
+			"example.com/?test=test",
+		},
+		{
+			"example.com{uri}/{~test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test",
+		},
+		{
+			"example.com{uri}/{>Test}",
+			"https://example.com/?test=test",
+			"example.com/?test=test/test-header",
+		},
+		{
+			"example.com{uri}/{?test}",
```

Comment:
Probably best to simplify the test. No need to both test {uri} and {?test} at the same time. Same for the above. Test one thing. Especially, when it's table tests, where duplication doesn't add more code, just test values.